ftw/file/imaging.py

```diff
@@ -0,0 +1,82 @@
+from plone.app.imaging.traverse import DefaultImageScaleHandler
+from zope.component import adapts
+from ftw.file.interfaces import IFtwFileField
+from plone.app.imaging import monkey
+from plone.app.imaging.interfaces import IImageScaleHandler
+from Products.Archetypes.interfaces.base import IBaseUnit
+from Products.Archetypes.utils import shasattr
+
+
+class FtwFileImageScaleHandler(DefaultImageScaleHandler):
+    adapts(IFtwFileField)
+
+    def getScale(self, instance, scale):
+        if self.context.get(instance).content_type.startswith('image/'):
+            return super(FtwFileImageScaleHandler, self).getScale(instance, scale)
+        else:
+            return None
+
+
+class ImagingMixin(object):
+
+    swallowResizeExceptions = False
+
+    def getSize(self, instance, scale=None):
+        """ get size of scale or original """
+        if scale is None:
+            return self.getUnwrapped(instance).getSize()
+        handler = IImageScaleHandler(self, None)
```

Comment:
@jone yes it would probably fix our problem but wouldn't be the risk of sideeffects be high since we could miss interfaces it currently provides?